Branch-SDK/Branch-SDK/BNCSpotlightService.m

```diff
@@ -0,0 +1,481 @@
+            } else if (spotlightCallback) {
+                spotlightCallback(url, spotlightIdentifier, nil);
+            }
+        }
+    }else {
+        
+        [self indexUsingSearchableItem:indexingParams
+                         thumbnailData:thumbnailData
+                              callback:callback
+                     spotlightCallback:spotlightCallback];
+    }
+#endif
+}
+
+-(NSMutableDictionary*) getSpotlightLinkDataWithCanonicalId: (NSString*)canonicalId
+                                                      title:(NSString *)title
+                                                description:(NSString *)description
+                                                   userInfo:(NSDictionary *)userInfo
+                                              typeOrDefault:(NSString*)typeOrDefault
+                                          publiclyIndexable:(BOOL)publiclyIndexable
+                                                   keywords:(NSSet*)keywords {
+    
+    NSMutableDictionary *spotlightLinkData = [[NSMutableDictionary alloc] init];
+    if (canonicalId) {
+        spotlightLinkData[BRANCH_LINK_DATA_KEY_CANONICAL_IDENTIFIER] = canonicalId;
+    }
+    spotlightLinkData[BRANCH_LINK_DATA_KEY_TITLE] = title;
+    spotlightLinkData[BRANCH_LINK_DATA_KEY_PUBLICLY_INDEXABLE] = @(publiclyIndexable);
+    spotlightLinkData[BRANCH_LINK_DATA_KEY_TYPE] = typeOrDefault;
+    
+    if (userInfo) {
+        [spotlightLinkData addEntriesFromDictionary:userInfo];
+    }
+    
+    // Default the OG Title, Description, and Image Url if necessary
+    if (!spotlightLinkData[BRANCH_LINK_DATA_KEY_OG_TITLE]) {
+        spotlightLinkData[BRANCH_LINK_DATA_KEY_OG_TITLE] = title;
+    }
+    
+    if (description) {
+        spotlightLinkData[BRANCH_LINK_DATA_KEY_DESCRIPTION] = description;
+        if (!spotlightLinkData[BRANCH_LINK_DATA_KEY_OG_DESCRIPTION]) {
+            spotlightLinkData[BRANCH_LINK_DATA_KEY_OG_DESCRIPTION] = description;
+        }
+    }
+    if (keywords) {
+        spotlightLinkData[BRANCH_LINK_DATA_KEY_KEYWORDS] = [keywords allObjects];
+    }
+    
+    return spotlightLinkData;
+}
+
+#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 90000
+-(CSSearchableItemAttributeSet *)getAttributeSetFromCanonicalId:(NSString *)canonicalId
+                                                          title:(NSString *)title
+                                                    description:(NSString *)description
+                                                           type:(NSString *)type
+                                                   thumbnailUrl:(NSURL *)thumbnailUrl
+                                                  thumbnailData:(NSData *)thumbnailData
+                                                       userInfo:(NSDictionary *)userInfo
+                                                       keywords:(NSSet *)keywords {
+    
+    if([CSSearchableItemAttributeSet class]) {
```

Comment:
you're going to need to convert these to reflection as we used to have in this file https://github.com/BranchMetrics/ios-branch-deep-linking/blob/master/Branch-SDK/Branch-SDK/BNCContentDiscoveryManager.m#L467